packages/demo/src/components/Login/index.js

```diff
@@ -1,35 +1,149 @@
 import React from 'react'
-import {connect} from 'cerebral/react'
+import { connect } from 'cerebral/react'
 import translations from '../../computed/translations'
+import LangSelector from '../LangSelector'
 
 export default connect(
   {
-    t: translations
+    t: translations,
+    $email: 'user.signIn.$email',
+    $password: 'user.signIn.$password',
+    $validationsErrors: 'user.signIn.$validationsErrors',
+    $error: 'user.signIn.$error'
   },
-  function Login ({t}) {
+  {
+    fieldChanged: 'user.fieldChanged',
+    signInClicked: 'user.signInClicked',
+    createUserClicked: 'user.createUserClicked'
+  },
+  function Login ({
+    t,
+    // state
+    $email,
+    $password,
+    $validationsErrors,
+    $error,
+    // signals
+    fieldChanged,
+    signInClicked,
+    createUserClicked
+  }) {
+    function getField ({
+      field,
+      icon,
+      value,
+      placeholder,
+      fieldType,
+      message
+    }) {
+      return (
+        <p className='control has-icon'>
+          <input className='input'
+            type={fieldType}
+            placeholder={placeholder}
+            value={value}
+            onChange={e => fieldChanged({
+              field,
+              value: e.target.value
```

Comment:
I would hash and salt (with email) password here, before it goes anywhere (just my own paranoid way).